components/director/internal/destinationcreator/service.go

```diff
@@ -107,42 +107,36 @@ func NewService(
 	}
 }
 
-// CreateDesignTimeDestinations is responsible to create so-called design time(destinationcreator.AuthTypeNoAuth) destination resource in the DB as well as in the remote destination service
-func (s *Service) CreateDesignTimeDestinations(ctx context.Context, destinationDetails operators.Destination, formationAssignment *model.FormationAssignment, depth uint8, skipSubaccountValidation bool) error {
-	subaccountID := destinationDetails.SubaccountID
+// CreateDesignTimeDestinations is responsible to create so-called design time destination resource in the DB as well as in the remote destination service
+func (s *Service) CreateDesignTimeDestinations(ctx context.Context, destinationDetails operators.DestinationRaw, formationAssignment *model.FormationAssignment, depth uint8, skipSubaccountValidation bool) error {
+	subaccountID := destinationDetails.GetSubaccountID()
 	region, err := s.getRegionLabel(ctx, subaccountID)
 	if err != nil {
 		return errors.Wrapf(err, "while getting region label for tenant with ID: %s", subaccountID)
 	}
 
-	destinationName := destinationDetails.Name
+	destinationName := destinationDetails.GetName()
 	strURL, err := buildDestinationURL(ctx, s.config.DestinationAPIConfig, URLParameters{
 		EntityName:   destinationName,
 		Region:       region,
 		SubaccountID: subaccountID,
-		InstanceID:   destinationDetails.InstanceID,
+		InstanceID:   destinationDetails.GetInstanceID(),
 	}, false)
 	if err != nil {
 		return errors.Wrapf(err, "while building destination URL")
 	}
 
-	destReqBody := &NoAuthDestinationRequestBody{
-		BaseDestinationRequestBody: BaseDestinationRequestBody{
-			Name:                 destinationName,
-			URL:                  destinationDetails.URL,
-			Type:                 destinationcreatorpkg.Type(destinationDetails.Type),
-			ProxyType:            destinationcreatorpkg.ProxyType(destinationDetails.ProxyType),
-			AuthenticationType:   destinationcreatorpkg.AuthType(destinationDetails.Authentication),
-			AdditionalProperties: destinationDetails.AdditionalProperties,
-		},
+	stripInternalFields, err := destinationDetails.StripInternalFields()
```

Comment:
Maybe `destinationWithStrippedInternalFields` or something that implies that this is still a destination? `destWithStrippedInternalFields`?